tutorials/advanced-development/01-advanced-pieces-topics/01-ajax-pieces.md

```diff
@@ -0,0 +1,134 @@
+
+## Filtering without reloading the page: using AJAX to enhance filters
+
+Apostrophe offers an general-purpose, extremely easy solution for AJAX refreshes.
+
+"What the heck is an AJAX refresh?" Right now, if you click on a filter, the whole page reloads. Instead, an AJAX refresh only updates the relevant part of the page.
+
+Add a `data-apos-ajax-context="name"` attribute to the outer div that should be refreshed when any link or form submission inside it takes place and has a URL that points back to the same page.
+
+_The value of the attribute must be unique on the page._
+
+Next, refactor your `index.html` template so that the actual list of people and any filters are in an `indexAjax.html` template, which is included at the appropriate point, wrapped in a div that has the `data-apos-ajax-context` attribute:
+
+{% code-tabs %}
+{% code-tabs-item title="lib/modules/people-pages/index.html" %}
+```markup
+{% extends "layout.html" %}
+<h2>People</h2>
+<div data-apos-ajax-context="people">
+  {% include "indexAjax.html" %}
+</div>
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+{% code-tabs %}
+{% code-tabs-item title="lib/modules/people-pages/indexAjax.html" %}
+```markup
+{% for piece in data.pieces %}
+  <h4>
+    {% set image = apos.images.first(piece.thumbnail) %}
+    {% if image %}
+      <img src="{{ apos.attachments.url(image, { size: 'one-sixth' }) }}" />
+    {% endif %}
+    <a href="{{ piece._url }}">{{ piece.title }}</a>
+  </h4>
+{% endfor %}
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+
+That's it! Really. And it automatically works with the filters that we saw earlier.
+
+**Tip:** you want to include your filter links and forms in `indexAjax.html` so that they too can be refreshed automatically, narrowing down the choices based on the other filters already in use. Any input elements or textareas that currently have the focus will not be refreshed, so if you are using form elements, you can even implement typeahead by triggering a submit of the form via JavaScript as the user types. (Keep in mind the accessibility consequences.)
+
+#### Combining a "Load More..." button with AJAX
+
+There's one catch with the ajax solution above: it doesn't yet account for a "Load More" button that appends to, rather than replacing, the current page's worth of content.
+
+But this isn't hard to accommodate. All you have to do is:
+
+1. Wrap a new element _inside_ your `data-apos-ajax-context` element around the content that makes up the current "page" of results. This should _not_ wrap around filter links or the "Load More" button itself.
+2. Give that element the `data-apos-ajax-append` attribute.
+3. Add `append=1` to the query string of your `Load More` button.
+
+Here's an example. Here we assume you already set up the `piecesFilters` option as described earlier in this tutorial to enable filtering people by tag.
```

Comment:
It's not earlier in this tutorial anymore, it's somewhere else presumably. Link to that place.

Reply:
Thanks. This is some of that "connective tissue" that I need to fix up, and it's tough to track it all down.